core/src/main/java/org/apache/calcite/sql/SqlOperator.java

```diff
@@ -192,8 +192,9 @@ public SqlOperandTypeChecker getOperandTypeChecker() {
    * @return acceptable range
    */
   public SqlOperandCountRange getOperandCountRange() {
-    if (operandTypeChecker != null) {
-      return operandTypeChecker.getOperandCountRange();
+    SqlOperandTypeChecker checker = getOperandTypeChecker();
```

Comment:
In case the sub class of SqlOperator override the `getOperandTypeChecker` function, but pass a null value of `operandTypeChecker` in constructor, use `getOperandTypeChecker` function instead of access `operandTypeChecker` directly.